src/client.js

```diff
@@ -3428,14 +3422,16 @@ MatrixClient.prototype.requestPasswordMsisdnToken = function(phoneCountry, phone
  * @return {module:client.Promise} Resolves: As requestEmailToken
  */
 MatrixClient.prototype._requestTokenFromEndpoint = function(endpoint, params) {
-    const id_server_url = url.parse(this.idBaseUrl);
-    if (id_server_url.host === null) {
-        throw new Error("Invalid ID server URL: " + this.idBaseUrl);
+    const postParams = Object.assign({}, params);
+
+    if (this.idBaseUrl) {
+        const idServerUrl = url.parse(this.idBaseUrl);
+        if (idServerUrl.host === null) {
```

Comment:
might be good to just check for falsey values in the event someone somehow creates a zero-length string as a host.
```suggestion
        if (!idServerUrl.host) {
```